src/app/modules/consultations/consultation-profile/consultation-questionnaire/consultation-questionnaire.component.ts

```diff
@@ -286,7 +280,8 @@ export class ConsultationQuestionnaireComponent implements OnInit, AfterViewInit
         variables:{
           userCount:{
             userId: this.currentUser.id,
-            profanityCount: this.profanityCount,
+            //TODO: Profanity filter feature, remove condition when ready fo deployment to production
+            profanityCount: !environment.production ? this.profanityCount: 0,
```

Comment:
If possible, we should not update the `profanityCount` at all in production. This ensures that we don't have to add conditions eveywhere this variable is used

Hypothetical example: Let's say there is a UI component that shows up only when `profanityCount` is greater than 0. If we miss adding the new condition there it will show up on prod. By disabling the updating at source, we don't have to worry about any side-effects related to it

Reply:
@gpalsingh The endpoints for this short nudge feature and profanity filter features are the same and `profanityCount` is a required field, therefore we must set the value for it.
I have reviewed the flow and this will not happen as we are setting it as 0 in every request and also not fetching the profane words list in the production environment i.e. no words are profane until these conditions are in place.

Reply:
@Skt2021 specifically I meant never increasing the value for `this.profanityCount`. If we never increase its value from `0` to any number then we don't have to add checks for it. I hope it makes sense

Reply:
@gpalsingh To do that we have to add this same condition to more places as the number of places where we are manipulation the `profanityCount` value will be greater than the API requests we are making.

Reply:
Makes sense. Just suggesting another minor improvement, to keep the code DRY, we can use [property getters](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/get#:~:text=See%20also-,getter,-The%20get%20syntax) here.

Reply:
@gpalsingh I looked into property getters but I have not been able to figure out how we can utilize that here. Can you please elaborate a little more? Thanks

Reply:
We can move the repeated code to the getter
```js
get profanityCountTemp() {
  return environment.production ? 0 : this.profanityCount;
}
```

Then use it later instead of repeating the whole expression
```js
const payload = {
  profanityCount: this.profanityCountTemp,
}
```

Reply:
@gpalsingh We have to write this getter in every object, where we have to use it. Right?

Reply:
@gpalsingh This is done.